src/main/java/com/iexec/resultproxy/challenge/Eip712ChallengeService.java

```diff
@@ -40,16 +48,15 @@ Eip712Challenge generateEip712Challenge(Integer chainId) {
     }
 
     private void saveEip712ChallengeString(String eip712ChallengeString) {
-        challengeId++;
-        challengeMap.put(challengeId, eip712ChallengeString);
+        challenges.put(eip712ChallengeString, eip712ChallengeString);
```

Comment:
We could use `null` or `new Object()` in the value to avoid unnecessary memory usage.

Reply:
My guess is that using `new Object()` would lead to more memory usage as it creates a new object whereas using the same key and value uses a single object and simply creates 2 references to it.
Using `null` would probably work, but I don't like the idea having a map with `null` values :thinking:

Reply:
What's the problem with having a null value (since we have a valid key)?

Reply:
I like "there's currently no out-of-the-box Collection with expiration" and your update with challenges.containsKey(). From that, I don't see clear problems we could have with null values.